### PR TITLE
Add Custom filter for lines

### PR DESCRIPTION
**Pull request by lilian-houdelet**

### CODE REVIEW

**Review by flomillot — CHANGES_REQUESTED** [review]
Countries does not work
https://github.com/gridsuite/gridexplore-app/blob/b38ba7e5225bdfb6151ff512e2505b2d5b903d5a/src/components/utils/rqb-inputs/value-editor.tsx#L24

File: src/components/utils/rqb-inputs/value-editor.tsx [hunk 1]
```diff
@@ -21,7 +21,13 @@ const ValueEditor = (props: ValueEditorProps) => {
         // No value needed for this operator
         return null;
     }
-    if (props.field === FieldType.COUNTRY) {
+    if (
```
Comment (flomillot) [inline src/components/utils/rqb-inputs/value-editor.tsx:24]: if (
        [FieldType.COUNTRY, FieldType.COUNTRY_1, FieldType.COUNTRY_2].includes(
            props.field as FieldType
        )
    )

Better than call toString() everytime, no ?

File: src/components/dialogs/filter/expert/expert-filter-utils.ts [hunk 2]
```diff
@@ -93,19 +101,36 @@ export const getOperators = (fieldName: string, intl: IntlShape) => {
     return defaultOperators;
 };
 
+function changeValueUnit(value: any, field: FieldType) {
+    if (microUnits.includes(field as FieldType)) {
```
Comment (flomillot) [inline src/components/dialogs/filter/expert/expert-filter-utils.ts]:
```suggestion
    if (microUnits.includes(field)) {
```

File: src/components/dialogs/filter/expert/expert-filter-utils.ts [hunk 3]
```diff
@@ -93,19 +101,36 @@ export const getOperators = (fieldName: string, intl: IntlShape) => {
...
 ): RuleGroupTypeExport {
     function transformRule(rule: CustomRuleType): RuleTypeExport {
         const isValueAnArray = Array.isArray(rule.value);
+
+        let updatedValue = rule.value;
+
+        if (rule.operator !== OperatorType.EXISTS) {
+            updatedValue = changeValueUnit(rule.value, rule.field as FieldType);
+        }
+
```
Comment (flomillot) [inline src/components/dialogs/filter/expert/expert-filter-utils.ts]: Why this code ?

File: src/components/dialogs/filter/expert/expert-filter-utils.ts [hunk 4]
```diff
@@ -93,19 +101,36 @@ export const getOperators = (fieldName: string, intl: IntlShape) => {
...
         return {
             field: rule.field as FieldType,
             operator: Object.values(OPERATOR_OPTIONS).find(
                 (operator) => operator.name === rule.operator
             )?.customName as OperatorType,
             value:
                 !isValueAnArray && rule.operator !== OperatorType.EXISTS
-                    ? rule.value
+                    ? updatedValue
```
Comment (flomillot) [inline src/components/dialogs/filter/expert/expert-filter-utils.ts]:
```suggestion
                    ? changeValueUnit(rule.value, rule.field as FieldType)
```
Why not just this ?

File: src/components/dialogs/filter/expert/expert-filter-utils.ts [hunk 5]
```diff
@@ -93,19 +101,36 @@ export const getOperators = (fieldName: string, intl: IntlShape) => {
...
                     : undefined,
             values: isValueAnArray ? rule.value : undefined,
```
Comment (flomillot) [inline src/components/dialogs/filter/expert/expert-filter-utils.ts]: What about `values` ?

File: src/components/dialogs/filter/expert/expert-filter-utils.ts [hunk 6]
```diff
@@ -140,13 +165,19 @@ export function importExpertRules(
             // values is a Set on server side, so need to sort
             if (rule.dataType === DataType.NUMBER) {
                 return rule.values
-                    .map((value) => parseFloat(value as string))
+                    .map((value) => {
```
Comment (flomillot) [inline src/components/dialogs/filter/expert/expert-filter-utils.ts]: You can keep the previous code and add a new map right after, then we avoid duplicate code.

**Review by flomillot — CHANGES_REQUESTED** [review]

File: src/components/dialogs/filter/expert/expert-filter-utils.ts (hunk 3 above)
Comment (lilian-houdelet) [reply src/components/dialogs/filter/expert/expert-filter-utils.ts:125]: we don't have any values for the operator EXISTS

Comment (lilian-houdelet) [reply src/components/dialogs/filter/expert/expert-filter-utils.ts]: Not mandatory with the changes

File: src/components/dialogs/filter/expert/expert-filter-utils.ts (hunk 5 above)
Comment (lilian-houdelet) [reply src/components/dialogs/filter/expert/expert-filter-utils.ts]: I forgot, I'll update it

**Review by flomillot — APPROVED** [review]

**Pull request merged**